### PR TITLE
Fix #948 - Correct DS DM19 call

### LINKED ISSUE #948: DS Query Not Shown in log for failure outcome
DS query not shown in log for failure outcome. ... it is presumed not sent, if not in the log.

Global data would be sent by TP.BAM, which would fit a TP interference model.

Log File Name: C:\CdriveSave\new j1939-84\v307 786 Samsara Link2.j1939-84

Start Test 1.7 - DM19: Calibration information

12:07:12.8353 Global DM19 Request
12:07:12.8379 18EAFFF9 [3] 00 D3 00 (TX)
12:07:13.4371 Timeout - No Response
FAIL: 6.1.7.2.a - Total number of reported CAL IDs is < user entered value for number of emission or diagnostic critical control units
FAIL: 6.1.7.5.c - OBD ECU Engine #1 (0) did not provide a response to Global query and did not provide a NACK for the DS query

End Test 1.7 - DM19: Calibration information

### LINKED ISSUE #949: Engine Global DM2 Response Believed to Have Been Missed
Engine Global DM2 Response Believed to Have Been Missed
Check Logs to confirm Engine Omission in Test 1.16

Log File Name: C:\CdriveSave\new j1939-84\v307 786 Samsara Link2.j1939-84

Start Test 1.16 - DM2: Previously Active DTCs

12:07:33.5834 Global DM2 Request
12:07:33.5866 18EAFFF9 [3] CB FE 00 (TX)
12:07:33.5873 18FECB17 [8] 00 FF 00 00 00 00 FF FF
DM2 from Instrument Cluster #1 (23): MIL: off, RSL: off, AWL: off, PL: off, No DTCs
12:07:33.5879 18FECB21 [8] 00 FF 0A 08 0E 07 FF FF
DM2 from Body Controller (33): MIL: off, RSL: off, AWL: off, PL: off
DTC 2058:14 - Source Address 58 , Special Instructions - 7 times
12:07:33.5897 18FECB05 [8] 00 FF 00 00 00 00 FF FF
DM2 from Shift Console - Primary (5): MIL: off, RSL: off, AWL: off, PL: off, No DTCs
12:07:33.5917 18FECB03 [8] 00 FF 00 00 00 00 FF FF
DM2 from Transmission #1 (3): MIL: off, RSL: off, AWL: off, PL: off, No DTCs
12:07:33.5946 18FECB37 [8] 00 FF 00 00 00 00 FF FF
DM2 from Lighting - Operator Controls (55): MIL: off, RSL: off, AWL: off, PL: off, No DTCs
12:07:33.5962 18FECB98 [8] 00 FF 00 00 00 00 FF FF
DM2 from Reserved for future assignment 152 (152): MIL: off, RSL: off, AWL: off, PL: off, No DTCs
12:07:33.5971 18FECB97 [8] 00 FF 00 00 00 00 FF FF
DM2 from Reserved for future assignment 151 (151): MIL: off, RSL: off, AWL: off, PL: off, No DTCs
12:07:33.5989 18FECB0B [8] C3 FF 00 00 00 00 FF FF
DM2 from Brakes - System Controller (11): MIL: not supported, RSL: off, AWL: off, PL: not supported, No DTCs
12:07:33.6179 18FECB19 [8] 00 FF 00 00 00 00 FF FF
DM2 from Passenger-Operator Climate Control #1 (25): MIL: off, RSL: off, AWL: off, PL: off, No DTCs

12:07:34.2100 Destination Specific DM2 Request to Engine #1 (0)
12:07:34.2139 18EA00F9 [3] CB FE 00 (TX)
12:07:34.2159 18FECB00 [8] 00 FF 00 00 00 00 FF FF
DM2 from Engine #1 (0): MIL: off, RSL: off, AWL: off, PL: off, No DTCs
**_FAIL: 6.1.16.4.b - OBD ECU Engine #1 (0) did not provide a response to Global query and did not provide a NACK for the DS query_**
// SAMSARA Influence? ECM supported DM2 with a single frame response; TP.DT should not be a factor?
// Global response for DM2 query received in Part 3
// Unexpected Service Tool notice #932 not observed

End Test 1.16 - DM2: Previously Active DTCs

Start Test 3.7 - DM2: Previously active DTCs

12:13:41.9131 Global DM2 Request
12:13:41.9186 18EAFFF9 [3] CB FE 00 (TX)
12:13:41.9197 18FECB03 [8] 00 FF 00 00 00 00 FF FF
DM2 from Transmission #1 (3): MIL: off, RSL: off, AWL: off, PL: off, No DTCs
12:13:41.9206 18FECB21 [8] 00 FF 0A 08 0E 07 FF FF
DM2 from Body Controller (33): MIL: off, RSL: off, AWL: off, PL: off
DTC 2058:14 - Source Address 58 , Special Instructions - 7 times
12:13:41.9209 18FECB17 [8] 00 FF 00 00 00 00 FF FF
DM2 from Instrument Cluster #1 (23): MIL: off, RSL: off, AWL: off, PL: off, No DTCs
12:13:41.9235 18FECB37 [8] 00 FF 00 00 00 00 FF FF
DM2 from Lighting - Operator Controls (55): MIL: off, RSL: off, AWL: off, PL: off, No DTCs
12:13:41.9231 18FECB2A [8] 00 FF 00 00 00 00 FF FF
DM2 from Headway Controller (42): MIL: off, RSL: off, AWL: off, PL: off, No DTCs
12:13:41.9238 18FECB05 [8] 00 FF 00 00 00 00 FF FF
DM2 from Shift Console - Primary (5): MIL: off, RSL: off, AWL: off, PL: off, No DTCs
**_12:13:41.9246 18FECB00 [8] 00 FF 00 00 00 00 FF FF
DM2 from Engine #1 (0): MIL: off, RSL: off, AWL: off, PL: off, No DTCs_**
12:13:41.9281 18FECB97 [8] 00 FF 00 00 00 00 FF FF
DM2 from Reserved for future assignment 151 (151): MIL: off, RSL: off, AWL: off, PL: off, No DTCs
12:13:41.9294 18FECB98 [8] 00 FF 00 00 00 00 FF FF
DM2 from Reserved for future assignment 152 (152): MIL: off, RSL: off, AWL: off, PL: off, No DTCs
12:13:41.9330 18FECBEC [8] 00 FF 3F F5 F3 09 FF FF
DM2 from Door Controller #1 (236): MIL: off, RSL: off, AWL: off, PL: off
DTC 521535:19 - Manufacturer Assignable SPN, Received Network Data In Error - 9 times
12:13:41.9350 18FECBED [8] 00 FF 45 F5 F3 09 FF FF
DM2 from Door Controller #2 (237): MIL: off, RSL: off, AWL: off, PL: off
DTC 521541:19 - Manufacturer Assignable SPN, Received Network Data In Error - 9 times
12:13:41.9374 18FECB0B [8] C3 FF 00 00 00 00 FF FF
DM2 from Brakes - System Controller (11): MIL: not supported, RSL: off, AWL: off, PL: not supported, No DTCs
12:13:41.9439 18FECB19 [8] 00 FF 00 00 00 00 FF FF
DM2 from Passenger-Operator Climate Control #1 (25): MIL: off, RSL: off, AWL: off, PL: off, No DTCs

12:13:42.5250 Destination Specific DM2 Request to Engine #1 (0)
12:13:42.5279 18EA00F9 [3] CB FE 00 (TX)
12:13:42.5346 18FECB00 [8] 00 FF 00 00 00 00 FF FF
DM2 from Engine #1 (0): MIL: off, RSL: off, AWL: off, PL: off, No DTCs

### PR DESCRIPTION
**Pull request by mmschaefer**
Fix #948 - Update the controller to actually do a DS call to the modules and corrected to test to expect results created by the correct call.

### CODE REVIEW

File: src/org/etools/j1939_84/controllers/part01/Part01Step07Controller.java [hunk 1]
```diff
@@ -220,10 +220,10 @@ protected void run() throws Throwable {
 
         // 6.1.7.4.a. Destination Specific (DS) DM19 to each OBD ECU (plus all
         // ECUs that responded to global DM19).
```
Comment (gumbelmj) [inline src/org/etools/j1939_84/controllers/part01/Part01Step07Controller.java:223]: This collection of addresses needs to also include the addresses of the modules that responded to the global request.

**Review by gumbelmj — CHANGES_REQUESTED** [review]
Need to also request packets from those that responded to the global request

**Review by gumbelmj — APPROVED** [review]

**Review by battjt — COMMENTED** [review]
How does this handle "Destination Specific (DS) DM19 to each OBD ECU **(plus all ECUs that responded to global DM19).**"?

**Comment by gumbelmj** [comment]
```
Stream.concat(getDataRepository().getObdModuleAddresses().stream(),
                                       globalPackets.stream().map(ParsedPacket::getSourceAddress))
                               .distinct()
```
Merges together the addresses from the OBD Modules addresses and the addresses from the modules that responding to the global request.

**Pull request merged**